extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcRecorder.java

```diff
@@ -132,6 +132,15 @@ private TenantConfigContext createTenantContext(Vertx vertx, OidcTenantConfig oi
             options.setClientSecretParameterName(null);
         }
 
+        // Proxy config
+        OidcTenantConfig.Proxy proxy = oidcConfig.getProxy();
+        if (proxy.host.isPresent() && proxy.port.isPresent()) {
+            options.setProxyOptions(proxy.asProxyOptions());
+        } else if (proxy.host.isPresent() || proxy.port.isPresent()) {
+            throw new ConfigurationException(
+                    "Both 'proxy.host' and 'proxy.port' must be configured to enable the proxy usage.");
+        }
+
```

Comment:
Yes, it would help to have optional configuration groups :-). But we can simplify a bit and ignore everything unless `proxy.host` is set, and only check the port inside of that condition
```
        if (proxy.host.isPresent()) {
           if (!proxy.port.isPresent()) {
               throw new ConfigurationException("Both 'proxy.host' and 'proxy.port' must be configured to enable the proxy usage.");
        }
        options.setProxyOptions(proxy.asProxyOptions());
   } 
```
How does it look :-) ?

Reply:
Oh, yeah. True also! And it looks more compact and nice. 😊